tests/tests/regression/issue_3748.rs

```diff
@@ -0,0 +1,52 @@
+use wgpu_test::{gpu_test, GpuTestConfiguration};
+
+use wgpu::*;
+
+/// Previously, for every user-defined vertex output a fragment shader had to have a corresponding
+/// user-defined input. This would generate `StageError::InputNotComsumed`.
+///
+/// This requirement was removed from the WebGPU spec. Now, when generating hlsl, wgpu will
+/// automatically remove any user-defined outputs from the vertex shader that are not present in
+/// the fragment inputs. This is necessary for generating correct hlsl:
+/// https://github.com/gfx-rs/naga/issues/1945
+#[gpu_test]
+static ALLOW_INPUT_NOT_CONSUMED: GpuTestConfiguration =
```

Comment:
This test might not be that necessary, unless an error will somehow be generated on DX12 when there is a mistake in the HLSL generation of the inter-stage interfaces. I'm not familiar with how failure for the interfaces to match manifests there.

Reply:
I also got around to testing it and saw an error :slightly_smiling_face: